### ISSUE TITLE
consider adding support for the alpha dropout

### ISSUE THREAD

**Issue by jvo203** [ml-explore/mlx#3178]
The normal dropout is already supported but there doesn't seem to be any support for the alpha dropout.

AI summary:

> Alpha Dropout is a specialized regularization technique for neural networks using Scaled Exponential Linear Units (SELU) activation functions. Unlike standard dropout, it randomly sets activations to the negative saturation value (−λω) rather than zero, maintaining the input's mean and variance (self-normalizing property) to prevent exploding/vanishing gradients.

**Comment by angeloskath** [comment]
That sounds great but I don't think it needs to be in MLX core. It should be trivial to implement in any library that needs it. I will close but if you think there is anything preventing an implementation of alpha dropout feel free to open another issue.

**Comment by jvo203** [comment]
Forgive me but it does not seem to be possible using the existing dropout implementation inside the mlx. If you look at the existing Dropout class, there is no way to specify a non-zero dropout value. That's why I raised this issue.

mlx.nn.Dropout
class Dropout(p: [float](https://docs.python.org/3/library/functions.html#float) = 0.5)
Randomly zero a portion of the elements during training.

The remaining elements are multiplied with 
1
1−𝑝 

 where 
𝑝
 is the probability of zeroing an element. This is done so the expected value of a given element will remain the same.

Parameters
:
p ([float](https://docs.python.org/3/library/functions.html#float)) – The probability to zero an element

**Comment by angeloskath** [comment]
Yes I know what I meant is you can simply write an `AlphaDropout` module in the project that you need it in. It should be a few lines and since it is not a ubiquitous module I don't think we should add it to MLX core.

I haven't tested it but sth like the following for instance

```python
import mlx.core as mx
import mlx.nn as nn

class AlphaDropout(nn.Module):
    def __init__(self, p: float = 0.5):
        super().__init__()
        if p < 0 or p > 1:
            raise ValueError(f"Dropout probability must be in [0, 1], got {p}")
        self.p = p

        # SELU negative saturation constant: -lambda * alpha
        self.alpha = -1.7580993408473766

    def __call__(self, x):
        if not self.training or self.p == 0.0:
            return x

        keep_prob = 1.0 - self.p

        mask = mx.random.uniform(shape=x.shape) < keep_prob
        y = mx.where(mask, x, self.alpha)

        # Affine correction to preserve mean=0, var=1
        a = (keep_prob + self.alpha**2 * keep_prob * (1 - keep_prob)) ** -0.5
        b = -a * self.alpha * (1 - keep_prob)

        return a * y + b
```

**Comment by jvo203** [comment]
OK, thank you. Did not know one could create own custom layers (i.e. a custom dropout) in mlx. I assumed it was all "baked-in" into the mlx APIs.
